frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/active-health-checks.spec.ts

```diff
@@ -0,0 +1,63 @@
+import { checkErrors } from '../../../integration-tests-cypress/support';
+import { MINUTE } from '../consts';
+
+describe('Test Popover behaviour for different active health check cases.', () => {
+  before(() => {
+    cy.login();
+    cy.visit('/');
+    cy.install();
+    cy.visit('/ocs-dashboards/block-file');
+  });
+
+  after(() => {
+    checkErrors();
+    cy.logout();
+  });
+
+  const scaleDownAndOpenHC = (state: string, deployments: string[]) => {
+    const scaleTo = (replicas: number) =>
+      deployments.forEach((deployment) => {
+        cy.exec(`oc scale --replicas=${replicas} deploy ${deployment} -n openshift-storage`);
+      });
+    scaleTo(0);
+    cy.byTestID(`${state}-secondary-status`, { timeout: 2 * MINUTE }).should('be.visible');
+    cy.byTestID('Storage Cluster-health-item')
+      .contains('Storage Cluster')
+      .click();
+    return () => scaleTo(1);
```

Comment:
```suggestion
  deployments.forEach(deployment =>  scaleDeployment(deployment, replicas));
```
Very odd pattern used here

Reply:
I've used a [closure pattern (lexical scoping)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Closures#lexical_scoping) to save the state of `scaleTo` when the argument is `1` so that I could easily invoke that from the calling parent function on demand.

https://github.com/openshift/console/pull/9690/files#diff-fa20d6bd8341e9a45df0d338a1cee9a6490b5f86dfe10f011c4c6e03b61c63b2R43

Reply:
ok, i get it but i don't seem to see why we need this additional complexity here. We can just scale down the pods here itself or do it in the tests scope, it will be much better